internal/ci/workflows.cue

```diff
@@ -46,134 +46,106 @@ test: _#bashWorkflow & {
 	name: "Test"
 	on: {
 		push: {
-			branches: ["main"]
-			tags: [_#releaseTagPattern]
+			branches: ["**"] // any branch (including '/' namespaced branches)
+			"tags-ignore": [_#releaseTagPattern]
 		}
-		pull_request: branches: ["**"]
+		pull_request: {}
 	}
 
 	jobs: {
+		start: {
+			"runs-on": _#linuxMachine
+			steps: [...(_ & {if: "${{ \(_#isCLCITestBranch) }}"})]
+			steps: [
+				_#writeNetrcFile,
+				_#startCLBuild,
+			]
+		}
 		test: {
+			needs:     "start"
 			strategy:  _#testStrategy
 			"runs-on": "${{ matrix.os }}"
 			steps: [
+				_#writeNetrcFile,
 				_#installGo,
-				_#checkoutCode,
+				_#checkoutCode & {
+					// "pull_request" builds will by default use a merge commit,
+					// testing the PR's HEAD merged on top of the master branch.
+					// For consistency with Gerrit, avoid that merge commit entirely.
+					// This doesn't affect "push" builds, which never used merge commits.
+					with: ref: "${{ github.event.pull_request.head.sha }}"
+				},
+				_#earlyChecks,
 				_#cacheGoModules,
-				_#setGoBuildTags & {
-					_#tags: "long"
-					if:     "${{ \(_#isMain) }}"
+				_#step & {
+					if:  "${{ \(_#isDefaultBranch) }}"
+					run: "echo CUE_LONG=true >> $GITHUB_ENV"
 				},
 				_#goModVerify,
 				_#goGenerate,
 				_#goTest,
+				_#goCheck,
 				_#goTestRace & {
-					if: "${{ \(_#isMain) }}"
+					if: "${{ \(_#isDefaultBranch) }}"
 				},
 				_#staticcheck,
 				_#goModTidy,
 				_#checkGitClean,
 				_#installUnity,
 				_#runUnity,
+				_#failCLBuild,
 			]
 		}
-	}
-}
-
-dailycheck: _#bashWorkflow & {
-
-	name: "Daily check"
-	on: {
-		schedule: [{cron: "0 9 * * *"}]
-	}
-
-	jobs: {
-		test: {
-			strategy:  _#testStrategy
-			"runs-on": "${{ matrix.os }}"
+		mark_ci_success: {
+			"runs-on": _#linuxMachine
+			if:        "${{ \(_#isCLCITestBranch) }}"
+			needs:     "test"
 			steps: [
-				_#installGo,
-				_#checkoutCode,
-				_#installUnity,
-				_#runUnity,
+				_#writeNetrcFile,
+				_#passCLBuild,
 			]
 		}
-	}
-}
-
-dispatch: _#bashWorkflow & {
-	// These constants are defined by github.com/cue-sh/tools/cmd/cueckoo
-	_#runtrybot: "runtrybot"
-	_#mirror:    "mirror"
-	_#importpr:  "importpr"
-	_#unity:     "unity"
-
-	_#dispatchJob: _#job & {
-		_#type: string
-		if:     "${{ github.event.client_payload.type == '\(_#type)' }}"
-	}
-
-	name: "Repository Dispatch"
-	on: ["repository_dispatch"]
-
-	jobs: {
-		test: _#dispatchJob & {
-			_#type:    _#unity
-			strategy:  _#testStrategy
-			"runs-on": "${{ matrix.os }}"
+		delete_build_branch: {
+			"runs-on": _#linuxMachine
+			if:        "${{ \(_#isCLCITestBranch) && always() }}"
```

Comment:
interesting that github's syntax highlighting seems to think `if` here is a keyword.

Reply:
Yeh that will be a function of https://github.com/cue-sh/vscode-cue